### DISCUSSION TITLE
Mobile performance and PSI score

### DISCUSSION THREAD

**Discussion by osa911** [garronej/tss-react#102] [Q&A] [answered]
Hi, 
I have worked on migrating my current project on mui5 with JSS to TSS.
The latest version of TSS (v3) was not correctly working with NextJS+SSR and MUI cache and I decided to use version 4 beta3 which works fine in this case.
After the main work on migration was done I compared the PSI score from lighthouse and [pagespeed.web.dev](https://pagespeed.web.dev/) before migration and after.
And as a result for desktop, it was approximately the same but for mobile devices, it has down (45 -> 26)
Maybe somebody did the same work or knows why it happened - please explain to me.
I'm thinking to go back to JSS temporarily...

**Comment by garronej** [comment]
I think I found what is the issue.  
Can you confirm that the performance issue is resolve in `4.0.0-beta.4` ?

**Comment by osa911** [comment]
I was busy this day, back to you once check it on production
Btw, Thanks for your work!

EDITED: `4.0.0-beta.4` is still not available for load

**Comment by garronej** [comment]
Sorry @osa911,   
There was a problem with my CI workflow. `4.0.0-beta.4` is now available on NPM

**Comment by osa911** [comment]
I've compared the impact of v4.beta4 in the lighthouse with v4.beta3 - it gives up to 5 points on average.
Probably the main issue in another place...
I have no idea right now

**Comment by garronej** [comment, marked as answer]
Hi @osa911,  
Could you try with 4.0.0-beta.7?

**Comment by osa911** [comment]
Let me try it 😉

чт, 18 серп. 2022 р. о 17:38 Joseph Garrone ***@***.***> пише:

**Reply by garronej** [reply]
Thanks, I hopes it fixes it.

**Reply by garronej** [reply]
@osa911 any news?

**Reply by osa911** [reply]
Still in review...
I hope today will check in production

**Reply by osa911** [reply]
So it has a small improvement I think
For mobile +5-8
For desktop +10-15

**Reply by garronej** [reply]
Ok, thank you for reporting.  
Could you check with `4.0.0-beta.8` ?  

Thank you in advance,

**Reply by osa911** [reply]
I'll let you know

**Reply by osa911** [reply]
In synthetic tests, it looks the same maybe +1 point for mobile.
The main thing is not worse! Let me check it on production

**Reply by osa911** [reply]
On Production env, it works fine as well and has a good score as the version before

**Reply by garronej** [reply]
Thanks for the feedback.  
I'll try to benchmark and improve along the way.
